src/line.c

```diff
@@ -229,13 +233,33 @@ int ccli_line_parse(const char *line, char ***pargv)
 			}
 		}
 		*w = '\0';
-		argv[argc++] = arg;
-		argv[argc] = NULL;
-	}
 
-	*pargv = argv;
+		if (strcmp(arg, "&&") == 0) {
+			if (cmd_idx == 0) {
+				*pcmd_args = (struct command_args *) malloc(sizeof(struct command_args));
+			} else {
+				*pcmd_args = (struct command_args *) realloc(*pcmd_args, sizeof(struct command_args) * (cmd_idx + 1));
+			}
+			(*pcmd_args + cmd_idx)->argv = argv;
+			(*pcmd_args + cmd_idx)->argc = argc;
+			argc = 0;
+			argv = malloc(sizeof(*v)); // TODO what to do here?
+			++cmd_idx;
+		} else {
```

Comment:
My comment on the memory allocation handling...

You need to handle the `ENOMEM` case of `malloc()` and `realloc()`. Otherwise, you will end up getting a NULL pointer dereference when you meet the failure case.

Also, you can't `realloc()` the existing pointer without a temporary variable. Because if you fail on the `realloc()`, the old pointer will be overwritten with a NULL. That's a memory leak.

Example for `malloc()`:
```
void *p;
p = malloc(size);
if (!p) {
   // p is NULL, dereference is illegal.
   // Handle run out of memory here...
   exit(1); // or return, or whatever...
}

// p is legal to use here...
```

For `realloc()`, you need to preserve the old pointer:
```
void *a;
void *tmp;

a = malloc(1024);
if (!a) {
   // handle error
   exit(1);
}

tmp = realloc(a, 2048);
if (!tmp) {
   free(a); // `a` is still a valid pointer here
   exit(1);
}

a = tmp; // now `a` is resized successfully here!
```

Reply:
Thank you very much @ammarfaizi2 !!